ENIGMAsystem/SHELL/Audio_Systems/DirectSound/DSbasic.cpp

```diff
@@ -195,108 +192,211 @@ namespace enigma_user
+  effectParams.fRightDelay = rightdelay;
+  effectParams.lPanDelay = pandelay;
+  */
 }
 
 void sound_effect_flanger(int sound, float wetdry, float depth, float feedback, float frequency, long wave, float delay, long phase) {
+  /*
+  DSFXFlanger effectParams = { };
+  effectParams.fWetDryMix = wetdry;
+  effectParams.fDepth = depth;
+  effectParams.fFeedback = feedback;
+  effectParams.fFrequency = frequency;
+  effectParams.lWaveform = wave;
+  effectParams.fDelay = delay;
+  effectParams.lPhase = phase;
+  */
 }
 
 void sound_effect_gargle(int sound, unsigned rate, unsigned wave) {
+  /*
+  DSFXGargle effectParams = { };
+  effectParams.dwRateHz = rate;
+  effectParams.dwWaveShape = wave;
+  */
 }
 
 void sound_effect_reverb(int sound, float gain, float mix, float time, float ratio) {
+  /*
+  DSFXWavesReverb effectParams = { };
+  effectParams.fInGain = gain;
+  effectParams.fReverbMix = mix;
+  effectParams.fReverbTime = time;
+  effectParams.fHighFreqRTRatio = ratio;
+  */
+}
+
+void sound_effect_compressor(int sound, float gain, float attack, float release, float threshold, float ratio, float delay) {
+  /*
+  DSFXCompressor effectParams = { };
+  effectParams.fGain = gain;
+  effectParams.fAttack = attack;
+  effectParams.fRelease = release;
+  effectParams.fThreshold = threshold;
+  effectParams.fRatio = ratio;
+  effectParams.fPredelay = delay;
+  */
 }
 
-//const _GUID& sound_effect_types[8] = { GUID_DSFX_STANDARD_CHORUS, GUID_DSFX_STANDARD_ECHO, GUID_DSFX_STANDARD_FLANGER, GUID_DSFX_STANDARD_GARGLE, 
-	//GUID_DSFX_WAVES_REVERB, GUID_DSFX_STANDARD_COMPRESSOR, GUID_DSFX_STANDARD_PARAMEQ };
-	
+void sound_effect_equalizer(int sound, float center, float bandwidth, float gain) {
+  /*
+  DSFXParamEq effectParams = { };
+  effectParams.fCenter = center;
+  effectParams.fBandwidth = bandwidth;
+  effectParams.fGain = gain;
+  */
+}
+
+static const GUID sound_effect_guids[7] = { GUID_DSFX_STANDARD_CHORUS, GUID_DSFX_STANDARD_ECHO, GUID_DSFX_STANDARD_FLANGER, GUID_DSFX_STANDARD_GARGLE,
+	GUID_DSFX_WAVES_REVERB, GUID_DSFX_STANDARD_COMPRESSOR, GUID_DSFX_STANDARD_PARAMEQ };
+
 void sound_effect_set(int sound, int effect) {
-/*
-  HRESULT hr;
-  DWORD dwResults[1];  // One element for each effect.
- 
-  // Describe the effect.
-  DSEFFECTDESC dsEffect;
-  memset(&dsEffect, 0, sizeof(DSEFFECTDESC));
-  dsEffect.dwSize = sizeof(DSEFFECTDESC);
-  dsEffect.dwFlags = 0;
-  dsEffect.guidDSFXClass = sound_effect_types[effect];
- 
-  get_sound(snd, sound, 0);
-	
-  // Set the effect
-  if (SUCCEEDED(hr = snd->soundBuffer->SetFX(1, &dsEffect, dwResults)))
-  {
-    #ifdef DEBUG_MODE
-    switch (dwResults[0])
-    {
-      case DSFXR_LOCHARDWARE:
-        printf("Effect was placed in hardware.");
-        break;
-      case DSFXR_LOCSOFTWARE:
-        printf("Effect was placed in software.");
-        break;
-      case DSFXR_UNALLOCATED:
-        printf("Effect is not yet allocated to hardware or software.");
-        break;
+  size_t numOfEffects = 0;
```

Comment:
FYI, I have no idea what this function is, and no idea what the implementation is doing, and so I'm ignoring it. This seems to be a generalization of the effects you've added above, but all of those take parameters out the ass. Does this just use reasonable defaults for everything?

Reply:
In my honest opinion, Mark Overmars did it right by just making them a combination of flags. I have no fucking idea why DirectSound wanted a struct for each flag honestly, it's insane. But thanks, it definitely does the same as GM8, and clearly my programming skills have improved 😈